pkg/k8s/informers/informers.go

```diff
@@ -0,0 +1,167 @@
+package informers
+
+import (
+	"github.com/rs/zerolog/log"
+	smiTrafficAccessClient "github.com/servicemeshinterface/smi-sdk-go/pkg/gen/client/access/clientset/versioned"
+	smiTrafficSpecClient "github.com/servicemeshinterface/smi-sdk-go/pkg/gen/client/specs/clientset/versioned"
+	smiTrafficSplitClient "github.com/servicemeshinterface/smi-sdk-go/pkg/gen/client/split/clientset/versioned"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/cache"
+
+	configClientset "github.com/openservicemesh/osm/pkg/gen/client/config/clientset/versioned"
+	policyClientset "github.com/openservicemesh/osm/pkg/gen/client/policy/clientset/versioned"
+)
+
+type InformerCollectionOpt func(*InformerCollection)
+
+func NewInformerCollection(meshName string, stop <-chan struct{}, kubeClient kubernetes.Interface, smiTrafficSplitClient smiTrafficSplitClient.Interface,
```

Comment:
nit: personally not a fan of variadic args with such a long list of params, as it makes it hard to tell what the variadic args are. Consider separating the constructor, ie: New().WithOptions, or even having the informers themselves passed as options, ie:


`NewInformerCollection("osm", stop, WithK8sClient(k8sClient), WithSMIClient(smiClient))`, etc

Maybe not, just wanted to provoke the line of thinking here

Reply:
Then we could have an option, ie: DontRunInformers, or RunAllInformers, as we discussed